### PR TITLE
[NEW] introduce permission system into apps-engine

### PR DESCRIPTION
**Pull request by lolimay**
# What? :boat:

This PR tries to introduce Permission System into the apps-engine.

# Why? :thinking:

# Links :earth_americas:

https://app.clickup.com/t/98bbwn

# PS :eyes:

### CODE REVIEW

**Review by d-gubert — CHANGES_REQUESTED** [review]

File: src/definition/permission/IPermissionCheckers.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export interface IPermissionCheckers {
```
[the remark this answers is not in the document]
Comment (lolimay) [reply src/definition/permission/IPermissionCheckers.ts:1]: ah, makes sense. Thanks for the note!

File: src/definition/permissions/IPermission.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export interface IPermission {
+    name: string;
+    requierd?: boolean;
```
Comment (thassiov) [inline src/definition/permissions/IPermission.ts:3]: typo here ;)

File: src/server/AppManager.ts [hunk 3]
```diff
@@ -380,7 +413,9 @@ export class AppManager {
         return true;
     }
 
-    public async add(appPackage: Buffer, enable = true, marketplaceInfo?: IMarketplaceInfo): Promise<AppFabricationFulfillment> {
+    public async add(appPackage: Buffer, installationParameters: IAppInstallParameters): Promise<AppFabricationFulfillment> {
+        const { enable, marketplaceInfo, permissionsGranted } = installationParameters;
```
Comment (thassiov) [inline src/server/AppManager.ts:417]: `enable` now does not have the default value `true`, making newly added apps to be disabled after installation. I'm changing this :grin:

**Comment by thassiov** [comment]
LGTM so far

**Review by d-gubert — APPROVED** [review]

**Pull request merged**